Updates/0.3.3.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\QueuedTracking;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\QueuedTracking\Queue\Manager;
+use Piwik\Updater;
+use Piwik\Updates;
+
+class Updates_0_3_3 extends Updates
+{
+    /**
+     * @var Manager
+     */
+    protected $queueManager;
+
+    public function doUpdate(Updater $updater)
+    {
+        $settings = $this->getSettings();
+        $numQueueWorkers = $settings->numQueueWorkers->getValue();
+
+        if ($numQueueWorkers <= 1) {
+            return; // it is pointless to rewrite one queue to the same queue
+        }
+
+        $this->rewriteQueues($numQueueWorkers);
+    }
+
+    /**
+     * @return Settings
+     */
+    public function getSettings()
+    {
+        return StaticContainer::get('Piwik\Plugins\QueuedTracking\Settings');
+    }
+
+    private function rewriteQueues($numQueueWorkers)
+    {
+        // queues are numbered from 0 to (number of queues - 1)
+        $queueIdsToRewrite = range(0, $numQueueWorkers - 1);
+
+        // loading all requests and removing them from queues
+        $requests = $this->loadRequestsFromQueues($queueIdsToRewrite);
```

Comment:
Here you can run into memory issues. Is it possible to rewrite the requests one at a time?